packages/smart-vaults/src/utils/OperatorManager.sol

```diff
@@ -0,0 +1,199 @@
+        0x0216194314f4301bb656e2100094647329591429172ab63bdb3b133cea9bbf00;
+
+    /* -------------------------------------------------------------------------- */
+    /*                                   STRUCTS                                  */
+    /* -------------------------------------------------------------------------- */
+
+    /// @notice Operator Manager storage structure.
+    struct OperatorManagerStorage {
+        mapping(address => bool) isOperator;
+    }
+
+    /* -------------------------------------------------------------------------- */
+    /*                                   EVENTS                                   */
+    /* -------------------------------------------------------------------------- */
+
+    /// @notice Event emitted when an operator is added.
+    event OperatorAdded(address indexed operator);
+
+    /// @notice Event emitted when an operator is removed.
+    event OperatorRemoved(address indexed operator);
+
+    /// @notice Event emitted when an operator executes a call.
+    event ExecutedTxFromOperator(address indexed operator, Call call);
+
+    /* -------------------------------------------------------------------------- */
+    /*                                   ERRORS                                   */
+    /* -------------------------------------------------------------------------- */
+
+    /// @notice Thrown when caller is not an operator.
+    error OnlyOperator();
+
+    /// @notice Thrown when target of call is this account.
+    error TargetCannotBeAccount();
+
+    /* -------------------------------------------------------------------------- */
+    /*                                  MODIFIERS                                 */
+    /* -------------------------------------------------------------------------- */
+
+    modifier onlyOperator() {
+        if (!_getOperatorManagerStorage().isOperator[msg.sender]) revert OnlyOperator();
+        _;
+    }
+
+    /* -------------------------------------------------------------------------- */
+    /*                          EXTERNAL/PUBLIC FUNCTIONS                         */
+    /* -------------------------------------------------------------------------- */
+
+    /**
+     * @notice Adds operator to the allowlist.
+     *
+     * @dev Can only be called by this account.
+     *
+     * @param operator_ address of operator.
+     */
+    function addOperator(address operator_) public {
+        _authorize();
+
+        _addOperator(operator_);
+    }
+
+    /**
+     * @notice Adds `operator` to the allowlist and makes a call to the `setupContract` passing `data_`.
+     *
+     * @dev Can only be called by this account.
+     *
+     * @param operator_ address of operator.
+     * @param setupContract_ address of contract to call to setup operator.
+     * @param data_ data passed to the setupContract.
+     */
+    function addAndSetupOperator(address operator_, address setupContract_, bytes calldata data_) public {
```

Comment:
not sure why we'd need this but.. might as well make it payable and pass along `msg.value`?

Reply:
not sure why we'd need the add and setup operator function?

Reply:
will come back to this later